diff --git a/Dockerfile b/Dockerfile
index af8e166..ef4e161 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -27,6 +27,7 @@ ARG EXTRA_BUILD_ARGS=""
 ARG CGO_ENABLED=1
 ENV CGO_ENABLED=$CGO_ENABLED
 COPY . .
+RUN cp -f /go/pkg/mod/github.com/boltdb/bolt@v1.3.1/bolt_amd64.go /go/pkg/mod/github.com/boltdb/bolt@v1.3.1/bolt_loong64.go
 RUN GOOS=linux GOARCH=${TARGETARCH} \
     go build $EXTRA_BUILD_ARGS -trimpath \
       -ldflags="-s -w -extldflags '-static' \
